sale_purchase_drop_shipping_asx/models/purchase_order.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    sale_shipping_id = fields.Many2one(
+        'res.partner',
+        string='Sale Order\'s Delivery Address',
+        compute='_compute_sale_shipping_id',
+        store=True,
+    )
+
+    @api.multi
+    @api.depends('origin')
+    def _compute_sale_shipping_id(self):
```

Comment:
Does this still work when origin is False?  Isn't it safer to find the value through PO line > destination move > procurement group > SO (the system allows users to override origin field)?